cpp/src/link_analysis/gunrock_hits.cu

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * ---------------------------------------------------------------------------*
+ * @brief wrapper calling gunrock's HITS analytic
+ * --------------------------------------------------------------------------*/
+
+#include <algorithms.hpp>
+#include <graph.hpp>
+
+#include <utilities/error_utils.h>
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <thrust/for_each.h>
+
+#include <gunrock/gunrock.h>
+
+namespace cugraph {
+
+namespace gunrock {
+
+template <typename vertex_t, typename edge_t, typename weight_t>
+void hits(cugraph::experimental::GraphCSRView<vertex_t, edge_t, weight_t> const &graph,
+          int max_iter,
+          weight_t tolerance,
+          weight_t const *starting_value,
+          bool normalized,
+          weight_t *hubs,
+          weight_t *authorities)
+{
+  //
+  //  NOTE:  gunrock doesn't support tolerance parameter
+  //         gunrock doesn't support passing a starting value
+  //         gunrock doesn't support the normalized parameter
+  //
+  //  FIXME: gunrock uses a 2-norm, while networkx uses a 1-norm.
+  //         They will add a parameter to allow us to specify
+  //         which norm to use.
+  //
```

Comment:
no error checking?   I'm fine with error checking be a FIXME so that users can start testing out the value of HITS sooner, but eventually, it should validate that a graph was passed in

Reply:
Added a couple of error checks.

We should talk about, generally, what level of error checking we want.  When the C++ code was converted to use GraphCOOView/GraphCSRView/GraphCSCView, most of the old error checks for graph became irrelevant.

The old checks verified that graph pointer was valid, and that the graph had the correct type (COO, CSR or CSC) as expected by the algorithm.  There is no graph pointer, it is a reference.  The reference is to the expected type.  So the compiler automatically does those checks (the code won't compile if you don't pass a GraphCSRView to the call).